sdcm/results_analyze/test.py

```diff
@@ -0,0 +1,550 @@
+        for es_data_path, data_value in es_data.items():
+            es_data_path = es_data_path.split('.')
+            if es_data_path[0] == '_source':
+                es_data_path = es_data_path[1:]
+            es_data_path = '.'.join(es_data_path)
+            es_data_path = cls._escape_filter_key(es_data_path)
+            if isinstance(data_value, str) and es_data_path not in cls._es_field_indexes and data_value != '*':
+                filters.append(f'{es_data_path}.keyword: \"{data_value}\"')
+            elif isinstance(data_value, bool):
+                filters.append(f'{es_data_path}: {str(data_value).lower()}')
+            else:
+                filters.append(f'{es_data_path}: {data_value}')
+        return ' AND '.join(filters)
+
+    @staticmethod
+    def _escape_filter_key(filter_key):
+        return ES_LUCENE_ESCAPE_REGEXP.sub(r'\\\1', filter_key)
+
+    def _get_es_query_from_self(self, data_path_patterns: list):
+        """
+        Builds ES query from self parameters using list of data path patterns from data_path_patterns
+            data pattern can have some_attr.*, which tells to build query using every data point from attribute some_attr
+        """
+        output = self._get_es_data_path_and_values_from_patterns(data_path_patterns, flatten=True)
+        return self._get_es_query_from_es_data(output)
+
+    @classmethod
+    def get_by_params(cls, es_index=es_index, **params):
+        es_query = cls._get_es_query_from_instance_data(params)
+        filter_path = cls._get_es_filters()
+        es_data = ES().search(
+            index=es_index,
+            q=es_query,
+            filter_path=filter_path,
+            size=10000)
+        if not es_data:
+            return []
+        es_data = es_data.get('hits', {}).get('hits', {})
+        if not es_data:
+            return []
+        return [cls(es_data=es_test_data) for es_test_data in es_data]
+
+    @classmethod
+    def get_by_test_id(cls, test_id, es_index):
+        return cls.get_by_params(es_index=es_index, test_id=test_id)
+
+    @classmethod
+    def get_metric_class(cls, metric_path, default=__DEFAULT__):
+        if default is __DEFAULT__:
+            return get_class_by_path(cls, metric_path)
+        return get_class_by_path(cls, metric_path, default)
+
+    @staticmethod
+    def gen_kibana_dashboard_url(
+            dashboard_path="app/kibana#/dashboard/03414b70-0e89-11e9-a976-2fe0f5890cd0?_g=()"
+    ):
+        return "%s/%s" % (ES()._conf.get('kibana_url'), dashboard_path)  # pylint: disable=protected-access
+
+    def get_subtests(self):
+        return self.get_by_params(es_index=self.es_index, main_test_id=self.test_id, subtest_name='*')
+
+    def get_same_tests_query(self):
+        list_of_attributes = [
+            'loader_cluster.type',
+            'loader_cluster.gce_type',
+            'loader_cluster.nodes',
+            'db_cluster.type',
+            'db_cluster.gce_type',
+            'db_cluster.nodes'
```

Comment:
We need to add the job_name so we can make sure that results from other jobs won't mess our data.